numpy/lib/_function_base_impl.py

```diff
@@ -4429,126 +4297,65 @@ def quantile(a,
+        (1-g)*y[j] + g*y[j+1]
 
-        * if g > 0 ; then take j
-        * if g = 0 and index is odd ; then take j
-        * if g = 0 and index is even ; then take i
+    where the index ``j`` and coefficient ``g`` are the integral and
+    fractional components of ``q * (n-1)``, and ``n`` is the number of
+    elements in the sample.
 
-    interpolated_inverted_cdf:
-        method 4 of H&F [1]_.
-        This method gives continuous results using:
+    This a special case of Equation 1 of H&F [1]_. More generally,
 
-        * alpha = 0
-        * beta = 1
+    - ``j = (q*n + m - 1) // 1``, and
+    - ``g = (q*n + m - 1) % 1``,
 
-    hazen:
-        method 5 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 1/2
-        * beta = 1/2
-
-    weibull:
-        method 6 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 0
-        * beta = 0
-
-    linear:
-        method 7 of H&F [1]_.
-        This method gives continuous results using:
-
-        * alpha = 1
-        * beta = 1
-
-    median_unbiased:
-        method 8 of H&F [1]_.
-        This method is probably the best method if the sample
-        distribution function is unknown (see reference).
-        This method gives continuous results using:
-
-        * alpha = 1/3
-        * beta = 1/3
-
-    normal_unbiased:
-        method 9 of H&F [1]_.
-        This method is probably the best method if the sample
-        distribution function is known to be normal.
-        This method gives continuous results using:
-
-        * alpha = 3/8
-        * beta = 3/8
-
-    lower:
-        NumPy method kept for backwards compatibility.
-        Takes ``i`` as the interpolation point.
-
-    higher:
-        NumPy method kept for backwards compatibility.
-        Takes ``j`` as the interpolation point.
-
-    nearest:
-        NumPy method kept for backwards compatibility.
-        Takes ``i`` or ``j``, whichever is nearest.
+    where ``m`` is defined differently for each value of the ``method``
+    parameter.
 
-    midpoint:
-        NumPy method kept for backwards compatibility.
-        Uses ``(i + j) / 2``.
+    =============================== =============== ===============
+    ``method``                      # in H&F [1]_   ``m``
+    =============================== =============== ===============
+    ``interpolated_inverted_cdf``   4               ``0``
+    ``hazen``                       5               ``1/2``
+    ``weibull``                     6               ``q``
+    ``linear`` (default)            7               ``1 - q``
+    ``median_unbiased``             8               ``q/3 + 1/3``
+    ``normal_unbiased``             9               ``q/4 + 3/8``
+    =============================== =============== ===============
+
+    Note that indices ``j`` and ``j + 1`` are clipped to the range ``0`` to
+    ``n-1`` when the results of the formula would be beyond the allowed range
```

Comment:
```suggestion
    ``n - 1`` when the results of the formula would be outside the allowed range
```

Reply:
The first of these changes can be considered to be an objective correction because it would be more consistent with the previous line.

What is the rationale for the second? (Maybe to avoid repetition of the syllable "be"?)

Reply:
For me "beyond" covers the case where `j > n - 1`, and "outside" covers the case where `j < 0` or `j > n - 1`. (disclaimer: I am not a native English speaker)

Reply:
I'm not sure that this usage is inconsistent with the meaning of "beyond", but I agree with your assessment of the connotations/typical usage, so "outside" may be the better choice. I'll change it.